auth-api/src/auth_api/services/entity_mapping.py

```diff
@@ -0,0 +1,163 @@
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Service for managing Affiliation Mapping data."""
+from typing import List
+
+from flask import current_app
+from requests import HTTPError
+from sqlalchemy import and_, or_
+from structured_logging import StructuredLogging
+
+from auth_api.exceptions.exceptions import ServiceUnavailableException
+from auth_api.models import db
+from auth_api.models.affiliation import Affiliation as AffiliationModel
+from auth_api.models.dataclass import AffiliationSearchDetails
+from auth_api.models.entity import Entity
+from auth_api.models.entity_mapping import EntityMapping
+from auth_api.services.rest_service import RestService
+
+logger = StructuredLogging.get_logger()
+
+
+class EntityMappingService:
+    """Manages all aspect of Entity Mapping data.
+
+    This manages updating, retrieving Affiliations in Entity Mapping data via the EntityMapping model.
+    """
+
+    @staticmethod
+    def get_filtered_affiliations(org_id: int, search_details: AffiliationSearchDetails) -> List[AffiliationModel]:
+        """Get affiliations from DB based on priority mapping logic.
+        - If no filters are applied (initial load), show only 1 page with 100 results.
+        - If filters are applied grab all identifiers
+        """
+        paginate_for_non_search = not any(
+            [search_details.identifier, search_details.status, search_details.name, search_details.type]
+        )
+
+        query = (
+            db.session.query(AffiliationModel)
+            .join(Entity, Entity.id == AffiliationModel.entity_id)
+            .join(
+                EntityMapping,
+                or_(
+                    EntityMapping.business_identifier == Entity.business_identifier,
+                    and_(
+                        EntityMapping.business_identifier.is_(None),
+                        EntityMapping.bootstrap_identifier == Entity.business_identifier,
+                    ),
+                    and_(
+                        EntityMapping.business_identifier.is_(None),
+                        EntityMapping.bootstrap_identifier.is_(None),
+                        EntityMapping.nr_identifier == Entity.business_identifier,
+                    ),
+                ),
+            )
+            .filter(AffiliationModel.org_id == int(org_id or -1), Entity.is_loaded_lear.is_(True))
+        ).order_by(AffiliationModel.created.desc())
+
+        if paginate_for_non_search:
+            limit = search_details.limit
+            page = search_details.page
+            offset_value = (page - 1) * limit
+            query = query.offset(offset_value).limit(limit)
+        return query.all()
```

Comment:
`affiliation.entity.business_identifier` is the part used for grabbing all the identifiers